src/frame/header.rs

```diff
@@ -0,0 +1,342 @@
+            BlockSize::Max4MB => 4 * 1024 * 1024,
+        }
+    }
+}
+
+#[derive(Clone, Copy, PartialEq, Debug)]
+pub enum BlockMode {
+    /// Every block is compressed independently. The default.
+    Independent,
+    /// Blocks can reference data from previous blocks.
+    ///
+    /// Effective when the stream contains small blocks.
+    Linked,
+}
+
+impl Default for BlockMode {
+    fn default() -> Self {
+        BlockMode::Independent
+    }
+}
+
+/*
+From: https://github.com/lz4/lz4/blob/dev/doc/lz4_Frame_format.md
+
+General Structure of LZ4 Frame format
+-------------------------------------
+
+| MagicNb | F. Descriptor | Block | (...) | EndMark | C. Checksum |
+|:-------:|:-------------:| ----- | ----- | ------- | ----------- |
+| 4 bytes |  3-15 bytes   |       |       | 4 bytes | 0-4 bytes   |
+
+Frame Descriptor
+----------------
+
+| FLG     | BD      | (Content Size) | (Dictionary ID) | HC      |
+| ------- | ------- |:--------------:|:---------------:| ------- |
+| 1 byte  | 1 byte  |  0 - 8 bytes   |   0 - 4 bytes   | 1 byte  |
+
+__FLG byte__
+
+|  BitNb  |  7-6  |   5   |    4     |  3   |    2     |    1     |   0  |
+| ------- |-------|-------|----------|------|----------|----------|------|
+|FieldName|Version|B.Indep|B.Checksum|C.Size|C.Checksum|*Reserved*|DictID|
+
+__BD byte__
+
+|  BitNb  |     7    |     6-5-4     |  3-2-1-0 |
+| ------- | -------- | ------------- | -------- |
+|FieldName|*Reserved*| Block MaxSize |*Reserved*|
+
+Data Blocks
+-----------
+
+| Block Size |  data  | (Block Checksum) |
+|:----------:| ------ |:----------------:|
+|  4 bytes   |        |   0 - 4 bytes    |
+
+*/
+#[derive(Debug, Clone)]
+pub struct FrameInfo {
+    /// If set, includes the total uncompressed size of data in the frame.
+    pub content_size: Option<u64>,
+    /// The identifier for the dictionary that must be used to correctly decode data.
+    /// The compressor and the decompressor must use exactly the same dictionary.
+    ///
+    /// Note that this is currently unsupported and if set will cause de (De)Compressor to error.
+    pub dict_id: Option<u32>,
+    /// The maximum uncompressed size of each data block.
+    pub block_size: BlockSize,
+    /// The block mode.
+    pub block_mode: BlockMode,
+    /// If set, includes a checksum for each data block in the frame.
+    pub block_checksums: bool,
+    /// If set, includes a content checksum to verify that the full frame contents have been decoded correctly.
+    pub content_checksum: bool,
+}
+
+impl Default for FrameInfo {
+    fn default() -> Self {
+        Self {
+            content_size: None,
+            dict_id: None,
+            block_size: BlockSize::default(),
+            block_mode: BlockMode::default(),
+            block_checksums: false,
+            content_checksum: false,
+        }
+    }
+}
+
+impl FrameInfo {
+    pub(crate) fn read_size(input: &[u8]) -> Result<usize, Error> {
+        let mut required = 7;
```

Comment:
is the 7 MIN_FRAME_INFO_SIZE?

`let mut required = MIN_FRAME_INFO_SIZE;`

from the variable name, it's not clear what is required

Reply:
Yes, good catch.